pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslContinuousInputPartition.java

```diff
@@ -0,0 +1,172 @@
+      PslPartitionOffset currentOffset,
+      PullSubscriber<SequencedMessage> subscriber,
+      ScheduledExecutorService pullExecutorService) {
+    return new PslContinuousInputPartitionReader(
+        subscriptionPath, currentOffset, subscriber, pullExecutorService);
+  }
+
+  @Override
+  public InputPartitionReader<InternalRow> createContinuousReader(PartitionOffset offset) {
+    assert PslPartitionOffset.class.isAssignableFrom(offset.getClass())
+        : "offset is not assignable to PslPartitionOffset";
+
+    PslPartitionOffset pslOffset = (PslPartitionOffset) offset;
+    PslPartitionOffset currentOffset =
+        PslPartitionOffset.builder()
+            .partition(pslOffset.partition())
+            // The first message to read is startOffset + 1
+            .offset(Offset.of(pslOffset.offset().value() + 1))
+            .build();
+
+    BufferingPullSubscriber subscriber;
+    try {
+      subscriber =
+          new BufferingPullSubscriber(
+              // TODO(jiangmichael): Pass credentials settings here.
+              (consumer) ->
+                  SubscriberBuilder.newBuilder()
+                      .setSubscriptionPath(options.subscriptionPath())
+                      .setPartition(pslOffset.partition())
+                      .setContext(PubsubContext.of(Constants.FRAMEWORK))
+                      .setMessageConsumer(consumer)
+                      .build(),
+              options.flowControlSettings(),
+              SeekRequest.newBuilder()
+                  .setCursor(Cursor.newBuilder().setOffset(currentOffset.offset().value()).build())
+                  .build());
+    } catch (CheckedApiException e) {
+      throw new IllegalStateException(
+          "Unable to create PSL subscriber for " + startOffset.toString(), e);
+    }
+    return createPartitionReader(
+        options.subscriptionPath(),
+        currentOffset,
+        subscriber,
+        Executors.newSingleThreadScheduledExecutor());
+  }
+
+  @Override
+  public InputPartitionReader<InternalRow> createPartitionReader() {
+    return createContinuousReader(startOffset);
+  }
+
+  private static class PslContinuousInputPartitionReader
+      implements ContinuousInputPartitionReader<InternalRow> {
+
+    private final SubscriptionPath subscriptionPath;
+    private final PullSubscriber<SequencedMessage> subscriber;
+    private final BlockingDeque<SequencedMessage> messages = new LinkedBlockingDeque<>();
+    private PslPartitionOffset currentOffset;
+    private SequencedMessage currentMsg;
+
+    private PslContinuousInputPartitionReader(
+        SubscriptionPath subscriptionPath,
+        PslPartitionOffset currentOffset,
+        PullSubscriber<SequencedMessage> subscriber,
+        ScheduledExecutorService pullExecutorService) {
+      this.subscriptionPath = subscriptionPath;
+      this.currentOffset = currentOffset;
+      this.subscriber = subscriber;
+      this.currentMsg = null;
+      pullExecutorService.scheduleAtFixedRate(
```

Comment:
See below comment, since next() needs to be blocking, I steered away from edge trigger in next. Otherwise, if edge triggered, I will need to do a while empty loop with manual Thread.sleep inside next() which is a bad pattern.